src/Apps/Artwork/Components/ArtworkSidebar/ArtworkSidebarCommercialButtons.tsx

```diff
@@ -47,13 +53,34 @@ const SaleMessage: React.FC<SaleMessageProps> = ({ saleMessage }) => {
 }
 
 interface ArtworkSidebarCommercialButtonsProps {
-  artwork: ArtworkSidebarCommercialButtons_artwork$data
+  artwork: ArtworkSidebarCommercialButtons_artwork$key
+  me: ArtworkSidebarCommercialButtons_me$key
 }
 
-const ArtworkSidebarCommerialButtons: React.FC<ArtworkSidebarCommercialButtonsProps> = ({
-  artwork,
-}) => {
-  const { relayEnvironment, router, user } = useSystemContext()
+const THE_PAST = new Date(0).toISOString()
+
+export const ArtworkSidebarCommercialButtons: React.FC<ArtworkSidebarCommercialButtonsProps> = props => {
+  const artwork = useFragment(ARTWORK_FRAGMENT, props.artwork)
+  const me = useFragment(ME_FRAGMENT, props.me)
+
+  const partnerOfferVisibilityEnabled = useFeatureFlag(
+    "emerald_partner-offers-to-artwork-page"
+  )
+
+  // Get the first not-ended partner offer, if available
+  const partnerOffer =
+    (partnerOfferVisibilityEnabled &&
+      me?.partnerOffersConnection &&
+      extractNodes(me.partnerOffersConnection)[0]) ||
+    null
+
+  // Fall back to a definitely past value because the timer hook doesn't like nulls
+  const partnerOfferTimer = useTimer(partnerOffer?.endAt || THE_PAST)
```

Comment:
#notblocking: This usage of the timer is only applicable if offer is present. Ideally we should probably figure put how to not instantiate all this objecs if offer is not in a picture - that would probably avoid usage of this PAST hack. But I don't really see right away how to refactor it. Cool with trying this as a follow up.

Reply:
Right, I had the same thought. I can't call the hook conditionally so this felt like the easiest route for now that wouldn't require handling a lot of weird/optional values in other usages of this timer.